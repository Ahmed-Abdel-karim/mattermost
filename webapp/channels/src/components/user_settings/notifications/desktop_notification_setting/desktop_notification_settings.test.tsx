// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import type {ComponentProps} from 'react';

import Constants, {NotificationLevels} from 'utils/constants';

import type {SelectOptions} from './desktop_notification_settings';
import DesktopNotificationSettings, {
    getCheckedStateForDesktopThreads,
    getValueOfSendMobileNotificationWhenSelect,
    getCheckedStateForDissimilarDesktopAndMobileNotification,
    validNotificationLevels,
    getDefaultMobileNotificationLevelWhenUsedDifferent,
} from './desktop_notification_settings';

jest.mock('utils/notification_sounds', () => {
    const original = jest.requireActual('utils/notification_sounds');
    return {
        ...original,
        hasSoundOptions: jest.fn(() => true),
    };
});

describe('DesktopNotificationSettings', () => {
    const baseProps: ComponentProps<typeof DesktopNotificationSettings> = {
        active: true,
        updateSection: jest.fn(),
        onSubmit: jest.fn(),
        onCancel: jest.fn(),
        saving: false,
        error: '',
        setParentState: jest.fn(),
        areAllSectionsInactive: false,
        isCollapsedThreadsEnabled: false,
        desktopActivity: NotificationLevels.MENTION,
        pushActivity: NotificationLevels.MENTION,
        pushStatus: Constants.UserStatuses.OFFLINE,
        desktopThreads: NotificationLevels.ALL,
        pushThreads: NotificationLevels.ALL,
        sendPushNotifications: true,
        desktopAndMobileSettingsDifferent: false,
        sound: 'false',
        callsSound: 'false',
        selectedSound: 'Bing',
        callsSelectedSound: 'Dynamic',
        isCallsRingingEnabled: false,
    };

    test('should match snapshot, on max setting', () => {
        const wrapper = shallow(
            <DesktopNotificationSettings {...baseProps}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on max setting with sound enabled', () => {
        const props = {...baseProps, sound: 'true'};
        const wrapper = shallow(
            <DesktopNotificationSettings {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on max setting with Calls enabled', () => {
        const props = {...baseProps, isCallsRingingEnabled: true};
        const wrapper = shallow(
            <DesktopNotificationSettings {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on max setting with Calls enabled, calls sound true', () => {
        const props = {...baseProps, isCallsRingingEnabled: true, callsSound: 'true'};
        const wrapper = shallow(
            <DesktopNotificationSettings {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on min setting', () => {
        const props = {...baseProps, active: false};
        const wrapper = shallow(
            <DesktopNotificationSettings {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should call props.updateSection and props.onCancel on handleChangeForMinSection', () => {
        const props = {...baseProps, updateSection: jest.fn(), onCancel: jest.fn()};
        const wrapper = shallow<DesktopNotificationSettings>(
            <DesktopNotificationSettings {...props}/>,
        );

        wrapper.instance().handleChangeForMinSection('');
        expect(props.updateSection).toHaveBeenCalledTimes(1);
        expect(props.updateSection).toHaveBeenCalledWith('');
        expect(props.onCancel).toHaveBeenCalledTimes(1);
        expect(props.onCancel).toHaveBeenCalledWith();

        wrapper.instance().handleChangeForMinSection('desktop');
        expect(props.updateSection).toHaveBeenCalledTimes(2);
        expect(props.updateSection).toHaveBeenCalledWith('desktop');
        expect(props.onCancel).toHaveBeenCalledTimes(2);
        expect(props.onCancel).toHaveBeenCalledWith();
    });

    test('should call props.updateSection on handleChangeForMaxSection', () => {
        const props = {...baseProps, updateSection: jest.fn()};
        const wrapper = shallow<DesktopNotificationSettings>(
            <DesktopNotificationSettings {...props}/>,
        );

        wrapper.instance().handleChangeForMaxSection('');
        expect(props.updateSection).toHaveBeenCalledTimes(1);
        expect(props.updateSection).toHaveBeenCalledWith('');

        wrapper.instance().handleChangeForMaxSection('desktop');
        expect(props.updateSection).toHaveBeenCalledTimes(2);
        expect(props.updateSection).toHaveBeenCalledWith('desktop');
    });

    test('should match snapshot, on buildMaximizedSetting', () => {
        const wrapper = shallow<DesktopNotificationSettings>(
            <DesktopNotificationSettings {...baseProps}/>,
        );

        expect(wrapper.instance().buildMaximizedSetting()).toMatchSnapshot();

        wrapper.setProps({desktopActivity: NotificationLevels.NONE});
        expect(wrapper.instance().buildMaximizedSetting()).toMatchSnapshot();
    });

    test('should match snapshot, on buildMinimizedSetting', () => {
        const wrapper = shallow<DesktopNotificationSettings>(
            <DesktopNotificationSettings {...baseProps}/>,
        );

        expect(wrapper.instance().buildMinimizedSetting()).toMatchSnapshot();

        wrapper.setProps({desktopActivity: NotificationLevels.NONE});
        expect(wrapper.instance().buildMinimizedSetting()).toMatchSnapshot();
    });
});

describe('getCheckedStateForDesktopThreads', () => {
    test('should return false when desktopThreads is undefined', () => {
        expect(getCheckedStateForDesktopThreads('all')).toBe(false);
    });

    test('should return false if either of them is not for all threads', () => {
        expect(getCheckedStateForDesktopThreads('all', 'mention')).toBe(false);
        expect(getCheckedStateForDesktopThreads('all', 'none')).toBe(false);
        expect(getCheckedStateForDesktopThreads('all', 'default')).toBe(false);

        expect(getCheckedStateForDesktopThreads('none', 'all')).toBe(false);
        expect(getCheckedStateForDesktopThreads('mention', 'all')).toBe(false);
        expect(getCheckedStateForDesktopThreads('default', 'all')).toBe(false);
    });

    test('should return true if both of them are for all threads', () => {
        expect(getCheckedStateForDesktopThreads('all', 'all')).toBe(true);
    });
});

describe('getValueOfSendMobileNotificationWhenSelect', () => {
    test('When input is undefined it should return the last option', () => {
        expect(getValueOfSendMobileNotificationWhenSelect(undefined)).not.toBeUndefined();

        const result = getValueOfSendMobileNotificationWhenSelect(undefined) as SelectOptions;
        expect(result.value).toBe(Constants.UserStatuses.OFFLINE);
    });

    test('when input is defined but is not a valid option it should return the last option', () => {
        // We are purposely testing with an invalid value hence the 'any'
        expect(getValueOfSendMobileNotificationWhenSelect('invalid' as any)).not.toBeUndefined();

        const result = getValueOfSendMobileNotificationWhenSelect('invalid' as any) as SelectOptions;
        expect(result.value).toBe(Constants.UserStatuses.OFFLINE);
    });

    test('When input is a valid option it should return the same option', () => {
        expect(getValueOfSendMobileNotificationWhenSelect(Constants.UserStatuses.ONLINE)).not.toBeUndefined();

        const result = getValueOfSendMobileNotificationWhenSelect(Constants.UserStatuses.ONLINE) as SelectOptions;
        expect(result.value).toBe(Constants.UserStatuses.ONLINE);

        expect(getValueOfSendMobileNotificationWhenSelect(Constants.UserStatuses.AWAY)).not.toBeUndefined();

        const result2 = getValueOfSendMobileNotificationWhenSelect(Constants.UserStatuses.AWAY) as SelectOptions;
        expect(result2.value).toBe(Constants.UserStatuses.AWAY);
    });
});

describe('getCheckedStateForDissimilarDesktopAndMobileNotification', () => {
    test('should return false when desktop and push notification levels are different', () => {
        validNotificationLevels.forEach((desktopLevel) => {
            validNotificationLevels.forEach((mobileLevel) => {
                if (desktopLevel !== mobileLevel) {
                    expect(getCheckedStateForDissimilarDesktopAndMobileNotification(desktopLevel, mobileLevel)).toBe(false);
                }
            });
        });
    });

    test('should return true when desktop and push notification levels are the same', () => {
        validNotificationLevels.forEach((level) => {
            expect(getCheckedStateForDissimilarDesktopAndMobileNotification(level, level)).toBe(true);
        });
    });

    test('should return false when desktop and push notification levels are undefined', () => {
        expect(getCheckedStateForDissimilarDesktopAndMobileNotification(undefined as any, undefined as any)).toBe(false);
        expect(getCheckedStateForDissimilarDesktopAndMobileNotification('hello' as any, NotificationLevels.ALL)).toBe(false);
    });
});

describe('getDefaultMobileNotificationLevelWhenUsedDifferent', () => {
    test('if invalid input is provided it should return the default value', () => {
        expect(getDefaultMobileNotificationLevelWhenUsedDifferent('invalid' as any)).toBe(NotificationLevels.MENTION);
    });

    test('it should never return default notification level', () => {
        validNotificationLevels.forEach((level) => {
            expect(getDefaultMobileNotificationLevelWhenUsedDifferent(level)).not.toBe(NotificationLevels.DEFAULT);
        });
    });

    test('it should return the next level when input is valid', () => {
        // expect(getDefaultMobileNotificationLevelWhenUsedDifferent(NotificationLevels.NONE)).toBe(NotificationLevels.ALL);
        expect(getDefaultMobileNotificationLevelWhenUsedDifferent(NotificationLevels.DEFAULT)).toBe(NotificationLevels.ALL);
        expect(getDefaultMobileNotificationLevelWhenUsedDifferent(NotificationLevels.MENTION)).toBe(NotificationLevels.NONE);
    });
});
