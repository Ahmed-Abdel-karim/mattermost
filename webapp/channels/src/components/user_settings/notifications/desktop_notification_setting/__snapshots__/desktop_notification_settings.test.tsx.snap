// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNotificationSettings should match snapshot, on buildMaximizedSetting 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <React.Fragment>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send notifications for:"
              id="user.settings.notifications.desktopAndMobile.sendNotificationFor"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="All new messages"
                id="user.settings.notifications.desktopAndMobile.allActivity"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="mention"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Mentions, direct messages, and group messages"
                id="user.settings.notifications.desktopAndMobile.onlyMentions"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="none"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Nothing"
                id="user.settings.notifications.desktopAndMobile.never"
              />
            </label>
          </div>
        </fieldset>
      </React.Fragment>,
      <React.Fragment>
        <br />
        <div
          className="checkbox single-checkbox"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Use different settings for mobile notifications"
              id="user.settings.notifications.desktopAndMobile.diffentMobileNotification.title"
            />
          </label>
        </div>
      </React.Fragment>,
      <hr />,
      <React.Fragment>
        <hr />
        <label
          className="singleSelectLabel"
          htmlFor="pushNotificationSelectInput"
          id="pushNotificationLabel"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Send mobile notifications when I am:"
            id="user.settings.notifications.desktopAndMobile.pushNotification"
          />
        </label>
        <StateManager
          aria-labelledby="pushNotificationLabel"
          className="react-select singleSelect"
          classNamePrefix="react-select"
          clearable={false}
          components={
            Object {
              "IndicatorSeparator": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          inputId="pushNotificationSelectInput"
          isClearable={false}
          isSearchable={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Online, away or offline"
                  id="user.settings.push_notification.online"
                />,
                "value": "online",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Away or offline"
                  id="user.settings.push_notification.away"
                />,
                "value": "away",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Offline"
                  id="user.settings.push_notification.offline"
                />,
                "value": "offline",
              },
            ]
          }
          value={
            Object {
              "label": <Memo(MemoizedFormattedMessage)
                defaultMessage="Offline"
                id="user.settings.push_notification.offline"
              />,
              "value": "offline",
            }
          }
        />
      </React.Fragment>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on buildMaximizedSetting 2`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <React.Fragment>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send notifications for:"
              id="user.settings.notifications.desktopAndMobile.sendNotificationFor"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="All new messages"
                id="user.settings.notifications.desktopAndMobile.allActivity"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="mention"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Mentions, direct messages, and group messages"
                id="user.settings.notifications.desktopAndMobile.onlyMentions"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="none"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Nothing"
                id="user.settings.notifications.desktopAndMobile.never"
              />
            </label>
          </div>
        </fieldset>
      </React.Fragment>,
      <React.Fragment>
        <br />
        <div
          className="checkbox single-checkbox"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Use different settings for mobile notifications"
              id="user.settings.notifications.desktopAndMobile.diffentMobileNotification.title"
            />
          </label>
        </div>
      </React.Fragment>,
      <hr />,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on buildMinimizedSetting 1`] = `
<SettingItemMin
  describe={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="For mentions and direct messages, without sound"
      id="user.settings.notifications.desktop.mentionsNoSound"
    />
  }
  section="desktopAndMobile"
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on buildMinimizedSetting 2`] = `
<SettingItemMin
  describe={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Off"
      id="user.settings.notifications.off"
    />
  }
  section="desktopAndMobile"
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on max setting 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <React.Fragment>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send notifications for:"
              id="user.settings.notifications.desktopAndMobile.sendNotificationFor"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="All new messages"
                id="user.settings.notifications.desktopAndMobile.allActivity"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="mention"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Mentions, direct messages, and group messages"
                id="user.settings.notifications.desktopAndMobile.onlyMentions"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="none"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Nothing"
                id="user.settings.notifications.desktopAndMobile.never"
              />
            </label>
          </div>
        </fieldset>
      </React.Fragment>,
      <React.Fragment>
        <br />
        <div
          className="checkbox single-checkbox"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Use different settings for mobile notifications"
              id="user.settings.notifications.desktopAndMobile.diffentMobileNotification.title"
            />
          </label>
        </div>
      </React.Fragment>,
      <hr />,
      <React.Fragment>
        <hr />
        <label
          className="singleSelectLabel"
          htmlFor="pushNotificationSelectInput"
          id="pushNotificationLabel"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Send mobile notifications when I am:"
            id="user.settings.notifications.desktopAndMobile.pushNotification"
          />
        </label>
        <StateManager
          aria-labelledby="pushNotificationLabel"
          className="react-select singleSelect"
          classNamePrefix="react-select"
          clearable={false}
          components={
            Object {
              "IndicatorSeparator": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          inputId="pushNotificationSelectInput"
          isClearable={false}
          isSearchable={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Online, away or offline"
                  id="user.settings.push_notification.online"
                />,
                "value": "online",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Away or offline"
                  id="user.settings.push_notification.away"
                />,
                "value": "away",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Offline"
                  id="user.settings.push_notification.offline"
                />,
                "value": "offline",
              },
            ]
          }
          value={
            Object {
              "label": <Memo(MemoizedFormattedMessage)
                defaultMessage="Offline"
                id="user.settings.push_notification.offline"
              />,
              "value": "offline",
            }
          }
        />
      </React.Fragment>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on max setting with Calls enabled 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <React.Fragment>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send notifications for:"
              id="user.settings.notifications.desktopAndMobile.sendNotificationFor"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="All new messages"
                id="user.settings.notifications.desktopAndMobile.allActivity"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="mention"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Mentions, direct messages, and group messages"
                id="user.settings.notifications.desktopAndMobile.onlyMentions"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="none"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Nothing"
                id="user.settings.notifications.desktopAndMobile.never"
              />
            </label>
          </div>
        </fieldset>
      </React.Fragment>,
      <React.Fragment>
        <br />
        <div
          className="checkbox single-checkbox"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Use different settings for mobile notifications"
              id="user.settings.notifications.desktopAndMobile.diffentMobileNotification.title"
            />
          </label>
        </div>
      </React.Fragment>,
      <hr />,
      <React.Fragment>
        <hr />
        <label
          className="singleSelectLabel"
          htmlFor="pushNotificationSelectInput"
          id="pushNotificationLabel"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Send mobile notifications when I am:"
            id="user.settings.notifications.desktopAndMobile.pushNotification"
          />
        </label>
        <StateManager
          aria-labelledby="pushNotificationLabel"
          className="react-select singleSelect"
          classNamePrefix="react-select"
          clearable={false}
          components={
            Object {
              "IndicatorSeparator": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          inputId="pushNotificationSelectInput"
          isClearable={false}
          isSearchable={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Online, away or offline"
                  id="user.settings.push_notification.online"
                />,
                "value": "online",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Away or offline"
                  id="user.settings.push_notification.away"
                />,
                "value": "away",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Offline"
                  id="user.settings.push_notification.offline"
                />,
                "value": "offline",
              },
            ]
          }
          value={
            Object {
              "label": <Memo(MemoizedFormattedMessage)
                defaultMessage="Offline"
                id="user.settings.push_notification.offline"
              />,
              "value": "offline",
            }
          }
        />
      </React.Fragment>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on max setting with Calls enabled, calls sound true 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <React.Fragment>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send notifications for:"
              id="user.settings.notifications.desktopAndMobile.sendNotificationFor"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="All new messages"
                id="user.settings.notifications.desktopAndMobile.allActivity"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="mention"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Mentions, direct messages, and group messages"
                id="user.settings.notifications.desktopAndMobile.onlyMentions"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="none"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Nothing"
                id="user.settings.notifications.desktopAndMobile.never"
              />
            </label>
          </div>
        </fieldset>
      </React.Fragment>,
      <React.Fragment>
        <br />
        <div
          className="checkbox single-checkbox"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Use different settings for mobile notifications"
              id="user.settings.notifications.desktopAndMobile.diffentMobileNotification.title"
            />
          </label>
        </div>
      </React.Fragment>,
      <hr />,
      <React.Fragment>
        <hr />
        <label
          className="singleSelectLabel"
          htmlFor="pushNotificationSelectInput"
          id="pushNotificationLabel"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Send mobile notifications when I am:"
            id="user.settings.notifications.desktopAndMobile.pushNotification"
          />
        </label>
        <StateManager
          aria-labelledby="pushNotificationLabel"
          className="react-select singleSelect"
          classNamePrefix="react-select"
          clearable={false}
          components={
            Object {
              "IndicatorSeparator": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          inputId="pushNotificationSelectInput"
          isClearable={false}
          isSearchable={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Online, away or offline"
                  id="user.settings.push_notification.online"
                />,
                "value": "online",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Away or offline"
                  id="user.settings.push_notification.away"
                />,
                "value": "away",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Offline"
                  id="user.settings.push_notification.offline"
                />,
                "value": "offline",
              },
            ]
          }
          value={
            Object {
              "label": <Memo(MemoizedFormattedMessage)
                defaultMessage="Offline"
                id="user.settings.push_notification.offline"
              />,
              "value": "offline",
            }
          }
        />
      </React.Fragment>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on max setting with sound enabled 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <React.Fragment>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send notifications for:"
              id="user.settings.notifications.desktopAndMobile.sendNotificationFor"
            />
          </legend>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="all"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="All new messages"
                id="user.settings.notifications.desktopAndMobile.allActivity"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={true}
                onChange={[Function]}
                type="radio"
                value="mention"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Mentions, direct messages, and group messages"
                id="user.settings.notifications.desktopAndMobile.onlyMentions"
              />
            </label>
            <br />
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="none"
              />
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Nothing"
                id="user.settings.notifications.desktopAndMobile.never"
              />
            </label>
          </div>
        </fieldset>
      </React.Fragment>,
      <React.Fragment>
        <br />
        <div
          className="checkbox single-checkbox"
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Use different settings for mobile notifications"
              id="user.settings.notifications.desktopAndMobile.diffentMobileNotification.title"
            />
          </label>
        </div>
      </React.Fragment>,
      <hr />,
      <React.Fragment>
        <hr />
        <label
          className="singleSelectLabel"
          htmlFor="pushNotificationSelectInput"
          id="pushNotificationLabel"
        >
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Send mobile notifications when I am:"
            id="user.settings.notifications.desktopAndMobile.pushNotification"
          />
        </label>
        <StateManager
          aria-labelledby="pushNotificationLabel"
          className="react-select singleSelect"
          classNamePrefix="react-select"
          clearable={false}
          components={
            Object {
              "IndicatorSeparator": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          inputId="pushNotificationSelectInput"
          isClearable={false}
          isSearchable={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Online, away or offline"
                  id="user.settings.push_notification.online"
                />,
                "value": "online",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Away or offline"
                  id="user.settings.push_notification.away"
                />,
                "value": "away",
              },
              Object {
                "label": <Memo(MemoizedFormattedMessage)
                  defaultMessage="Offline"
                  id="user.settings.push_notification.offline"
                />,
                "value": "offline",
              },
            ]
          }
          value={
            Object {
              "label": <Memo(MemoizedFormattedMessage)
                defaultMessage="Offline"
                id="user.settings.push_notification.offline"
              />,
              "value": "offline",
            }
          }
        />
      </React.Fragment>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;

exports[`DesktopNotificationSettings should match snapshot, on min setting 1`] = `
<SettingItemMin
  describe={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="For mentions and direct messages, without sound"
      id="user.settings.notifications.desktop.mentionsNoSound"
    />
  }
  section="desktopAndMobile"
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Desktop and mobile notifications"
      id="user.settings.notifications.desktopAndMobile.title"
    />
  }
  updateSection={[Function]}
/>
`;
